File: 프로그래머스/Lv.1/정수 제곱근 판별/정수 제곱근 판별.c
#include <stdio.h>
#include <stdbool.h>
#include <stdlib.h>

long long solution(long long n) {
    long long answer = -1;
    if(n == 1){
        answer = 4;
    }
    for(long long i =2;i<=n/2;i++){
        if(i*i == n){
            answer = (i+1)*(i+1);
            break;
        }
    }
    return answer;
}

File: 프로그래머스/Lv.1/정수 제곱근 판별/정수 제곱근 판별.js
function solution(n) {
  let answer = -1;
  if(n === 1){
      answer =4;
  }
  for(let i=2;i<=n/2;i++){
      if(i*i === n){
          answer = (i+1) ** 2;
          break;
      }
  }
  return answer;
}
